viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala

```diff
@@ -778,23 +785,27 @@ class NodesAndJunctionsService(roadwayDAO: RoadwayDAO, roadwayPointDAO: RoadwayP
 
     time(logger, "Handling node point templates") {
       try {
-        val filteredLinks = projectLinks.filter(pl => RoadClass.forNodes.contains(pl.roadNumber.toInt) && pl.status != RoadAddressChangeType.Termination).filterNot(_.track == Track.LeftSide)
-        val groupSections = filteredLinks.groupBy(l => (l.roadNumber, l.roadPartNumber))
+        val filteredLinks = projectLinks.filter(pl =>
+          RoadClass.forNodes.contains(pl.roadNumber.toInt) && // Road number is < 20000 or between 40000-70000
+          pl.status != RoadAddressChangeType.Termination && // Terminated are handled elsewhere
+          pl.track != Track.LeftSide) // Nodes are created only for tracks 0 and 1 (Combined and Right side)
+
+        val groupedByRoadPart = filteredLinks.groupBy(pl => (pl.roadNumber, pl.roadPartNumber))
 
-        groupSections.values.foreach { group =>
-          val administrativeClassSections: Seq[Seq[ProjectLink]] = continuousNodeSections(group.sortBy(_.startAddrMValue))._2
+        groupedByRoadPart.values.foreach { projectLinks =>
+          val administrativeClassSections: Seq[Seq[ProjectLink]] = continuousNodeSections(projectLinks.sortBy(_.startAddrMValue))._2
           administrativeClassSections.foreach { section =>
 
             val headProjectLink = section.head
             val headReversed = roadwayChanges.exists(ch => ch.changeInfo.target.startAddressM.nonEmpty && headProjectLink.startAddrMValue == ch.changeInfo.target.startAddressM.get
               && ch.changeInfo.target.endAddressM.nonEmpty && headProjectLink.endAddrMValue == ch.changeInfo.target.endAddressM.get && ch.changeInfo.reversed)
 
-            val headNodePoint: Option[NodePoint] = mappedRoadwayNumbers.find { rl =>
+            val headNodePoint: Option[NodePoint] = projectRoadLinkChanges.find { rl =>
               headProjectLink.startAddrMValue == rl.newStartAddr && headProjectLink.endAddrMValue == rl.newEndAddr && headProjectLink.roadwayNumber == rl.newRoadwayNumber
             }.flatMap { rl =>
               if (headReversed) {
                 nodePointDAO.fetchRoadAddressNodePoints(Seq(rl.originalRoadwayNumber, headProjectLink.roadwayNumber))
-                  .find(np => np.beforeAfter == BeforeAfter.Before && np.addrM == rl.newStartAddr)
+                .find(np => np.beforeAfter == BeforeAfter.After && np.addrM == rl.newStartAddr)
```

Comment:
Hmm, tässä Before on vaihtunu Afteriks ja rivillä 825 toisinpäin - miks?
Oliks ne vaan ihan väärin aiemmin?

Reply:
Juu ne oli väärin